src/System.Private.CoreLib/shared/System/Type.Helpers.cs

```diff
@@ -131,7 +131,7 @@ public virtual Type[] FindInterfaces(TypeFilter filter, object filterCriteria)
             if (cnt == c.Length)
                 return c;
 
-            Type[] ret = new Type[cnt];
+            Type?[] ret = new Type[cnt];
```

Comment:
This will not contain null values, so this in theory shouldn't be needed. and the return type shouldn't need a `?` either.

The reason why this can't contain null values, is because in line 139 we do:
```cs
if (c[i] != null)
   ret[cnt++] = c[i];
```

Reply:
Then compiler complains "Possible null reference assignment" because we a setting c[i] = null on row 127.
```
    if (!filter(c[i], filterCriteria))
        c[i] = null;
```
I can use ! operator too, which one preferred?
```
    if (c[i] != null)
        ret[cnt++] = c[i]!;
```

Reply:
Let's use the `!` instead and add that to this issue linking this comment: https://github.com/dotnet/corefx/issues/36427

cc: @jaredpar

Reply:
updated and added bug to dotnet/corefx#36427